### PR TITLE
use_coveralls function

### LINKED ISSUE #680: coveralls support
It would be nice to have a `use_coveralls()` function to easily use [coveralls.io](https://coveralls.io/) in a package.

This can be easily accomplished now using [covr](https://github.com/jimhester/covr).

I will have a pull request implementing this shortly.

### PR DESCRIPTION
**Pull request by jimhester**
This implements the use_coveralls function to add a coveralls test
coverage report to the repository.  It also enables a coveralls report
for the devtools repository.

closes #680 

I did not include the roxygen results other than NAMESPACE in this pull request to make the diff easier to read.

Let me know if you want anything changed, or if my description is not clear.

You can see examples of repositories with coveralls enabled at

- <https://github.com/jimhester/covr>
- <https://github.com/kevinushey/rex>
- <https://github.com/jimhester/lintr>
- <https://github.com/jimhester/gmailr>

### CODE REVIEW

**Comment by jimhester** [comment]
Sorry, had a few issues with the official pull request,  dfdae79 should be good to go though, you will also need to enable the repo at <https://coveralls.io/repos/new>.

**Comment by hadley** [comment]
Do you have plans to submit covr to CRAN? Ideally, I'd like to rely only on CRAN packages in the released version

**Comment by jimhester** [comment]
I am but I am using some devtools functions which are unexported in the current release.  (https://github.com/jimhester/covr/issues/63).  So ideally I would like to submit `covr` to CRAN after the next devtools release.  Circular dependencies, so we would have to coordinate the releases somehow.

Alternatively I am currently working on extracting the `with_*` functions, `R`, `RCMD` functions from devtools into a new `robustr` package, along with a couple of utility functions I needed to write for `covr`.  `robustr` would provide safe alternatives for modifying global state and calling R from within R.  Once that is done I can convert `covr` and `devtools` to use that package for those functions and that would solve this issue.

If you have other suggestions on functions that you think would be useful to include in `robustr` I would be happy to include them.

**Comment by hadley** [comment]
Ok, that sounds good. In that case, I'm happy to merge this (after some review).

File: R/infrastructure.R [hunk 1]
```diff
@@ -142,6 +142,50 @@ use_travis <- function(pkg = ".") {
...
+  gh <- github_info(pkg)
+  message("Adding coveralls information into .travis.yml for ", pkg$package, ". Next: \n",
+    " * Turn on coveralls for this repo at https://coveralls.io/repos/new\n",
+    " * Add a coveralls shield to your README.md:\n",
+    "[![Coverage Status]",
+      "(https://img.shields.io/coveralls/", gh$username, "/", gh$repo, ".svg)]",
+      "(https://coveralls.io/r/", gh$username, "/", gh$repo, "?branch=master)"
+  )
+
+  install_loc <- grep("^install:$", travis_content)
```
Comment (hadley) [inline R/infrastructure.R]: I'm a bit leery about automatically modifying the travis file, especially now that there are two main formats. Could you please just print out instructions? (and update them for the `language: r` style)

**Comment by hadley** [comment]
Bump

**Comment by jimhester** [comment]
Ok I think everything should be ok with this now, I added a note to the NEWS as well.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**